foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/util/YamlUtils.java

```diff
@@ -193,6 +186,36 @@ public Collection<Object> values() {
         public Set<Map.Entry<String, Object>> entrySet() {
             return contents.entrySet();
         }
+
+        private <T> boolean hasValue(Class<T> type, String... path) {
+            YamlObject parent = getParentIfExists(path);
+            if (parent == null) {
+                return false;
+            }
+            String key = path[path.length - 1];
+            return parent.containsKey(key) && type.isInstance(parent.get(key));
+        }
+
+        private <T> T getValue(String... path) {
+            YamlObject parent = getParentIfExists(path);
+            if (parent == null) {
+                throw new NoSuchElementException("Parent object does not exist at " + Arrays.toString(path));
+            }
+            String key = path[path.length - 1];
+            return (T) parent.get(key);
```

Comment:
I: If we passed in a `Class` object for the type, we could provide a more helpful message when the type of the value doesn't match what's expected. E.g.:
```java
    if (!type.isInstance(parent.get(key)) {
        throw new IllegalArgumentException("Value for [" + key + "] is not of type " + type.getSimpleName());
    }
```

However, I guessed that in nearly every case the call site will contain the literal path to the value being accessed and so a couple extra steps to check the source code to find what value isn't of the expected type isn't that big of a deal and simplifies both the call site and the implementation of this method.